chore(readme): improve readme and add customers' showcase (#1792)
microsoft/react-native-test-app: README.md

[![build](https://github.com/microsoft/react-native-test-app/actions/workflows/build.yml/badge.svg?event=push)](https://github.com/microsoft/react-native-test-app/actions/workflows/build.yml)
[![npm version](https://img.shields.io/npm/v/react-native-test-app)](https://www.npmjs.com/package/react-native-test-app)

> - [Quick Start](#quick-start-⚡)

Reviewer: GitHub omits the emoji:
```suggestion
- [Quick Start](#quick-start-)
```

- [Migrate an Existing Test App](https://github.com/microsoft/react-native-test-app/wiki/Migrate-an-Existing-Test-App)
- [Configuring the Test App](https://github.com/microsoft/react-native-test-app/wiki/Manifest-%28app.json%29)

> - [Known Issues](#known-issues-⚠️)

Reviewer: GitHub converted this to unicode:
```suggestion
- [Known Issues](#known-issues-%EF%B8%8F)
```

React Native Test App (RNTA) provides test apps for all platforms as a package.
It handles the native bits for you so you can focus on what's important: your
product.

> ## Libraries Using RNTA 📚

Reviewer: As a potential consumer, I'd rather look at quick start first. I think we should move this further down. Maybe below "Learn More".

<table>
<tr>
<td>

> - Microsoft's
>   [fluentui-react-native](https://github.com/microsoft/fluentui-react-native)
> - Microsoft's [rnx-kit](https://github.com/microsoft/rnx-kit)
> - Shopify's [restyle](https://github.com/Shopify/restyle)

Reviewer: Nit: Restyling this a bit makes the table look more consistent:
```suggestion
- [microsoft/fluentui-react-native](https://github.com/microsoft/fluentui-react-native)
- [microsoft/rnx-kit](https://github.com/microsoft/rnx-kit)
- [shopify/restyle](https://github.com/Shopify/restyle)
```

Also, should this be three columns?

- [react-native-google-signin](https://github.com/react-native-google-signin/google-signin)
- [react-native-masked-view](https://github.com/react-native-masked-view/masked-view)
- [react-native-menu](https://github.com/react-native-menu/menu)

</td>
<td>

- [react-native-add-calendar-event](https://github.com/vonovak/react-native-add-calendar-event)
- [react-native-apple-authentication](https://github.com/invertase/react-native-apple-authentication)
- [react-native-blur](https://github.com/Kureev/react-native-blur)
- [react-native-image-editor](https://github.com/callstack/react-native-image-editor)
- [react-native-netinfo](https://github.com/react-native-netinfo/react-native-netinfo)
- [react-native-webview](https://github.com/react-native-webview/react-native-webview)
- [react-native-segmented-control](https://github.com/react-native-segmented-control/segmented-control)
- [sparkfabrik-react-native-idfa-aaid](https://github.com/sparkfabrik/sparkfabrik-react-native-idfa-aaid)

</td>
</tr>
</table>

...[and many more!](https://github.com/microsoft/react-native-test-app/network/dependents)

> _Are you using `react-native-test-app`? Submit a PR to add it to the list!_

Reviewer: Let's be consistent about using RNTA:

```suggestion
_Are you using RNTA? Submit a PR to add it to the list!_
```

In case we rename it to `react-native-template-app` later 😉

[![build](https://github.com/microsoft/react-native-test-app/actions/workflows/build.yml/badge.svg?event=push)](https://github.com/microsoft/react-native-test-app/actions/workflows/build.yml)
[![npm version](https://img.shields.io/npm/v/react-native-test-app)](https://www.npmjs.com/package/react-native-test-app)

> React Native Test App (RNTA) provides test apps for all platforms as a
> package. It handles the native bits for you so you can focus on what's

> > important: your product.

Reviewer: Can we unquote this? The rendering looks off with it.

[platform specific instructions](https://github.com/microsoft/react-native-test-app/wiki/Quick-Start#platform-specific-instructions)
in the wiki.

> ## Learn more 🔍

Reviewer: Title case to be consistent
```suggestion
## Learn More 🔍
```
